### PR TITLE
Fix rocks db configuration for txn test

### PR DESCRIPTION
**Pull request by wentaoxu**
use default configurations when call the function 'new_local_engine' if the caller use 'write_cf','lock_cf' or 'default_cf'.
@siddontang @BusyJay @disksing

### CODE REVIEW

File: src/util/rocksdb/mod.rs [hunk 1]
```diff
@@ -80,14 +80,19 @@ impl<'a> CFOptions<'a> {
     }
 }
 
-pub fn new_engine(path: &str, cfs: &[&str]) -> Result<DB, String> {
+pub fn new_engine(path: &str, cfs: &[&str], opts: Option<Vec<CFOptions>>) -> Result<DB, String> {
```
Comment (siddontang) [inline src/util/rocksdb/mod.rs]: can we pass `Option<&[CFOptions]>`

File: src/server/server.rs [hunk 2]
```diff
@@ -252,6 +252,7 @@ mod tests {
     }
 
     #[test]
+    // if this failed, unset the environmental variables 'http_proxy' and 'https_proxy', and retry.
```
Comment (siddontang) [inline src/server/server.rs]: @BusyJay 

Seem we should avoid resolving 127.0.0.1.

**Comment by disksing** [comment]
LGTM.

**Comment by siddontang** [comment]
Please sign the CLA @wentaoxu

**Comment by siddontang** [comment]
CI failed @wentaoxu

File: src/util/rocksdb/mod.rs (hunk 1 above)
Comment (wentaoxu) [reply src/util/rocksdb/mod.rs]: I think it doesn't matter now, because the caller don't need the options.

**Comment by wentaoxu** [comment]
@siddontang fix CI, my rustfmt is the wrong version.

**Comment by wentaoxu** [comment]
PTAL @siddontang @BusyJay @zhangjinpeng1987

File: src/util/rocksdb/mod.rs [hunk 3]
```diff
@@ -80,14 +80,19 @@ impl<'a> CFOptions<'a> {
...
     let mut db_opts = DBOptions::new();
     db_opts.enable_statistics(true);
-    let mut cfs_opts = Vec::with_capacity(cfs.len());
-    for cf in cfs {
-        cfs_opts.push(CFOptions::new(*cf, ColumnFamilyOptions::new()));
+    match opts {
```
Comment (zhangjinpeng1987) [inline src/util/rocksdb/mod.rs]: let cf_opts = match opts {
...
};
new_engine_opt(path, db_opts, cf_opts)

Comment (wentaoxu) [reply src/util/rocksdb/mod.rs]: ok

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/util/rocksdb/mod.rs (hunk 1 above)
Comment (siddontang) [reply src/util/rocksdb/mod.rs]: should keep the same style with cfs.

**Pull request merged**